components/email_aliases/resources/content/email_aliases_modal.tsx

```diff
@@ -260,8 +274,9 @@ export const EmailAliasModal = ({
       // wrong type in its JSDoc.
       // TODO(https://github.com/brave/brave-browser/issues/48960): fix the
       // JSDoc generation issue so that this cast is not needed.
+      const response = await emailAliasesService.generateAlias()
       const proposedEmail =
-        (await emailAliasesService.generateAlias()) as unknown as string
+        typeof response === 'string' ? response : response.result.success!
```

Comment:
Hey, I think this change is wrong - the generated result type is a `{ success: string, error: string }` but that's not actually what happens - the result type will either be `Promise.resolve(success)` or `Promise.reject(error)`. Unfortunately the generated types are wrong because we're using the `generate_legacy_js_bindings` flag.

I'm working on migrating us off it, but its non-trivial :laughing: